cmake/unittest.cmake

```diff
@@ -0,0 +1,106 @@
+set(UT_NAME ${PROJECT_NAME}_gtests)
+set(ONNX_ROOT ${PROJECT_SOURCE_DIR})
+
+find_package(Threads)
+
+function(add_whole_archive_flag lib output_var)
+  if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    set(${output_var} -Wl,-force_load,$<TARGET_FILE:${lib}> PARENT_SCOPE)
+  elseif(MSVC)
+    # In MSVC, we will add whole archive in default.
+    set(${output_var} -WHOLEARCHIVE:$<SHELL_PATH:$<TARGET_FILE:${lib}>> PARENT_SCOPE)
+  else()
+    # Assume everything else is like gcc
+    set(${output_var} "-Wl,--whole-archive $<TARGET_FILE:${lib}> -Wl,--no-whole-archive" PARENT_SCOPE)
+  endif()
+endfunction()
+
+set(${UT_NAME}_libs
+    ${googletest_STATIC_LIBRARIES}
+)
+
+if(NOT TARGET protobuf::libprotobuf)

```

Comment:
why do we need to find protobuf here again? I think we have already done this in the top level cmake file?

Reply:
I checked the last failure, it should be irrelevant. I think the `if` should be removed.